src/validation.cpp

```diff
@@ -5531,14 +5536,6 @@ bool ChainstateManager::PopulateAndValidateSnapshot(
     for (int i = AFTER_GENESIS_START; i <= snapshot_chainstate.m_chain.Height(); ++i) {
         index = snapshot_chainstate.m_chain[i];
 
-        // Fake nTx so that LoadBlockIndex() loads assumed-valid CBlockIndex
-        // entries (among other things)
-        if (!index->nTx) {
-            index->nTx = 1;
-        }
-        // Fake nChainTx so that GuessVerificationProgress reports accurately
```

Comment:
Would `GuessVerificationProgress` lose accuracy with the removal or was the comment wrong?

Reply:
> do you have a code path where nChainTx is read when it was wrong (previously) or is unset (now)

no, I never understood how the fake `nChainTx` values would help with guessing progress, but also wasn't 100% sure.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/29370#discussion_r1479169504

> unrelated to this pull: Maybe an `Assume(nchaintx)` can be added to `GuessVerificationProgress`?

I went ahead and added this in a separate commit